File: src/website/GetRequest.java
package website;

import com.sun.net.httpserver.Headers;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;

import java.io.IOException;
import java.io.OutputStream;
import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.HashMap;

public abstract class GetRequest implements HttpHandler {
    @Override
    public void handle(HttpExchange httpExchange) throws IOException {
        if (httpExchange.getRequestMethod().equals("GET")) {
            String query = httpExchange.getRequestURI().getQuery();

            HashMap<String, String> params = new HashMap<>();

            String[] res = query.split("&");
            for (String str : res) {
                String[] values = str.split("=");
                params.put(values[0], values[1]);

            }

            String response = myrequest(params);


            Headers h = httpExchange.getResponseHeaders();
            h.set("Content-Type", "application/json");
            httpExchange.sendResponseHeaders(200, 0);

            OutputStream os = httpExchange.getResponseBody();
            os.write(response.getBytes());
            os.close();
        }
    }

    /**
     *
     * @param params received get params from website
     * @return json reply to website
     */
    public abstract String myrequest(HashMap<String, String> params);

}

File: src/java/com/wasteinformationserver/website/basicrequest/GetRequest.kt
package com.wasteinformationserver.website.basicrequest

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import java.io.IOException
import java.util.*

/**
 * basic GET request handler
 * reply function has to be implemented!
 */
abstract class GetRequest : HttpHandler {
    @Throws(IOException::class)
    override fun handle(httpExchange: HttpExchange) {
        if (httpExchange.requestMethod == "GET") {
            val query = httpExchange.requestURI.query
            val params = HashMap<String, String>()
            val res = query.split("&".toRegex()).toTypedArray()
            for (str in res) {
                val values = str.split("=".toRegex()).toTypedArray()
                params[values[0]] = values[1]
            }
            val response = myrequest(params)
            val h = httpExchange.responseHeaders
            h["Content-Type"] = "application/json"
            httpExchange.sendResponseHeaders(200, 0)
            val os = httpExchange.responseBody
            os.write(response.toByteArray())
            os.close()
        }
    }

    /**
     * @param params received get params from com.wasteinformationserver.website
     * @return json reply to com.wasteinformationserver.website
     */
    abstract fun myrequest(params: HashMap<String, String>): String
}